src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { Todo } from '../../Types/Todo';
+import usersFromServe from '../../api/users';
+import { getUserById } from '../../services/user';
+import { getNewTodoId } from '../../services/id';
+
+type Props = {
+  onAdd: (todo: Todo) => void;
+  todos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onAdd, todos }) => {
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const resetForm = () => {
+    setTitle('');
+    setUserId(0);
+    setHasUserIdError(false);
+    setHasTitleError(false);
+  };
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onAdd({
+      id: getNewTodoId(todos),
+      title,
+      userId,
+      completed: false,
+      user: getUserById(userId),
```

Comment:
The `user` property should be prepared in the `App` component and passed to `TodoForm` as part of the todo object. This aligns with the requirement to prepare data in one place and pass it to child components.